libopenage/pathfinding/pathfinder.cpp

```diff
@@ -269,11 +269,11 @@ const Pathfinder::portal_star_t Pathfinder::portal_a_star(const PathRequest &req
 		}
 
 		// get the exits of the current node
-		const auto &exits = current_node->get_exits(current_node->entry_sector);
+		const auto &exits = current_node->get_exits(current_node->entry_sector.value());
```

Comment:
```suggestion
		ENSURE(current_node->entry_sector != std::nullopt, "Entry sector not set for portal node.");
		const auto &exits = current_node->get_exits(*current_node->entry_sector);
```

Doing it like this will check if the value of the optional was correctly set, but only in debug mode because the openage `ENSURE` macro only gets included in debug builds (like an assert). You don't need to add it here, I just wanted to mention it :)

Reply:
Thanks for the advice, i've added it.